core/src/test/scala/org/apache/spark/SparkContextSchedulerCreationSuite.scala

```diff
@@ -135,7 +135,7 @@ class SparkContextSchedulerCreationSuite
       assert(sched.getClass === Utils.classForName(expectedClassName))
     } catch {
       case e: SparkException =>
-        assert(e.getMessage.contains("YARN mode not available"))
+        assert(e.getMessage.contains("Could not parse Master URL: 'yarn'"))
```

Comment:
This test seems pretty outdated... it will always fall into this case (checking the exception message), which is even less interesting now than it was before.